python/lsst/daf/butler/butler.py

```diff
@@ -49,17 +55,95 @@ class Butler:
     ----------
     config : `Config`
         Configuration.
+    collection : `str` or `None`
+        Collection to use for all input lookups, overriding config['collection']
+        if provided.
+    run : `str`, `Run`, or `None`
+        Collection associated with the `Run` to use for outputs, overriding
+        config['run'].  If a `Run` associated with the given Collection does
+        not exist, it will be created.  If "collection" is None, this
+        collection will be used for input lookups as well; if not, it must have
+        the same value as "run".
+
+    Raises
+    ------
+    ValueError
+        Raised if neither 'collection' nor 'run' are provided by argument or
+        config, or if both are provided and are inconsistent.
     """
 
-    def __init__(self, config=None):
+    @staticmethod
+    def makeRepo(root, config=None, expand=False):
+        """Create an empty data repository by adding a butler.yaml config
+        to a repository root directory.
+
+        Parameters
+        ----------
+        root : `str`
+            Filesystem path to the root of the new repository.
+        config : `Config` or `None`
+            Configuration to write to the repository, after setting any
+            root-dependent Registry or Datastore config options.
+        expand : `bool`
```

Comment:
I still like `expand` better; I think brevity is more valuable for keyword argument names that regular local variables in terms of overall readability.  @timj, want to break the tie? :-)

Reply:
I think I'd prefer `withDefaults` to `expandDefaults`. You are asking for a fully qualified standalone configuration to be written, rather than the bare minimum based on what you gave in `config` and the root-specific entries. You aren't really expanding as such, you are filling it in to make it explicit. `complete` or `standalone` ?

Reply:
Ooh, I like `standalone`.